apps/yggdrasil/lib/yggdrasil/role_resource.ex

```diff
@@ -0,0 +1,13 @@
+defmodule Yggdrasil.RoleResource do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "roles_resources" do
+    belongs_to :role, Yggdrasil.Role
+    belongs_to :resource, Yggdrasil.Resource
+
+    has_many :role_resource_permissions, Yggdrasil.RoleResourcePermission
```

Comment:
I'm a little confused what's going on here. You have a `role` <-> `role resource` <-> `role resource permission`? Why not a single table to connect all of them? 

```elixir
role:
    name

resource:
    name

permission:
    name

role_permission:
    role_id
    resource_id
    permission_id
```